### PR TITLE
[20.01] Mount job_working_directory/configs rw

### PR DESCRIPTION
**Pull request by mvdbeek**
Many tools have this pattern where they move the config file somewhere,
which doesn't work if it is mounted read-only.

### CODE REVIEW

**Comment by bernt-matthias** [comment]
Is this related: https://github.com/galaxyproject/galaxy/pull/9236?

**Comment by mvdbeek** [comment]
I would say they're separate.

File: lib/galaxy/tool_util/deps/container_classes.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def add_var(name, value):
             if self.job_info.tool_directory:
                 defaults += ",$tool_directory:default_ro"
             if self.job_info.job_directory:
-                defaults += ",$job_directory:default_ro,$job_directory/outputs:rw"
+                defaults += ",$job_directory:default_ro,$job_directory/outputs:rw,$job_directory/configs"
```
Comment (jmchilton) [inline lib/galaxy/tool_util/deps/container_classes.py]: Should this end in ``:rw``?

**Comment by jmchilton** [comment]
Many tools? That is gross - it prevents the tools from being re-submittable. Any interest in limiting the profile version of this behavior?

File: lib/galaxy/tool_util/deps/container_classes.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_classes.py]: It should, I wonder why this seems to work regardless ?

**Comment by mvdbeek** [comment]
> Many tools? That is gross - it prevents the tools from being re-submittable. Any interest in limiting the profile version of this behavior?

Not that many (maybe 10 in all of the iuc), and it's usually the last step in the tool. We could also include the configs dir in https://github.com/galaxyproject/galaxy/pull/9245, that'd take care of that as well.

**Comment by mvdbeek** [comment]
Alright I made the configs:rw conditional on the tool profile and made this safe for resubmission.

**Pull request merged**